engine/utility/public/fileIO.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <chrono>
+#include <fstream>
+#include <optional>
+#include <string>
+#include <vector>
+
+/// @brief File interface system for opening and closing files
+namespace fileIO
+{
+constexpr auto DEFAULT_READ_FLAGS = std::ios::in | std::ios::binary;
+constexpr auto DEFAULT_WRITE_FLAGS = std::ios::out | std::ios::trunc | std::ios::binary;
+
+constexpr auto TEXT_READ_FLAGS = std::ios::in;
+constexpr auto TEXT_WRITE_FLAGS = std::ios::out | std::ios::trunc;
+
+using FileTime = std::chrono::time_point<std::chrono::file_clock>;
+
+/// <summary>
+/// Open a file stream for reading. Specify 0 or std::ios::flags
+/// </summary>
+std::optional<std::ifstream> OpenReadStream(const std::string& path,
```

Comment:
string_view preferred?

Reply:
string view requires a little more error handling but I can try it

Reply:
I was able to use string_view for the external API, but I still require an internal conversion back to string for opening the streams (this is generally safer, since stringviews are not guaranteed to be null terminated)
```c++
std::optional<std::ifstream> fileIO::OpenReadStream(std::string_view path,
    std::ios::openmode flags)
{
    std::ifstream stream(std::string(path), flags);
    // ....
}
```
I don't think this change is worth it since it can lead to weird bugs and file paths need to be null terminated, which will either force checking logic on our end or creating an std::string out of the view